### DISCUSSION TITLE
Demodoctrine module not compatible with Prestashop 8

### DISCUSSION THREAD

**Discussion by rflorent** [PrestaShop/PrestaShop#33237] [Q&A]
Hello,

I was looking for a module example to learn the right way and have a fast start.
I lost hours trying to make this module works with version 8...
Please add some compatibility information per module...

"This module demonstrates how to use Doctrine entities in PrestaShop 1.7.7 and above."
That's wrong.

Thanks

**Comment by leemyongpakvn** [comment]
I think we need to discuss more at https://github.com/PrestaShop/example-modules/issues. Some detailed warnings during your install process will be helpful to debug :bug:

**Reply by MatShir** [reply]
Should we tranfert the discussion in https://github.com/PrestaShop/example-modules/issues repo ? @leemyongpakvn

**Reply by rflorent** [reply]
When I try to create an issue, it redirect to discussion

**Reply by leemyongpakvn** [reply]
@MatShir Certainly. But I have only one option `PrestaShop/hummingbird` when trying to **Transfer this disscussion** from my side.

**Reply by MatShir** [reply]
For now, let's keep here ! Unless if we found a committer who wants to own/manage this repo. To get back to the subject, it misses some details to understand the issue

**Reply by kpodemski** [reply]
Example modules are supposed to be discussed here. It's on purpose.

**Reply by leemyongpakvn** [reply]
In this case, Install from Back Office in Module Catalog does **not** provide useful information to debug, Using the command `php ./bin/console prestashop:module install demodoctrine` does :D
In terminal, I see 2 issues, one with write permission to /var/cache directory, and one with missing class.
![demodoctrine_install_write_permission](https://github.com/PrestaShop/PrestaShop/assets/3759923/5333faa0-c380-4461-a142-21d211515070)
![demodoctrine_install_missing_class](https://github.com/PrestaShop/PrestaShop/assets/3759923/cd882132-66b3-4e78-b35b-cae1f738442b)
We can solve these 2 issues with `chown chmod`, and <s>`composer dumpautoload`</s> `composer install` respectively. After that, the installation will go smoothly, then the module works as supposed in both FO and BO.

**Comment by kpodemski** [comment]
Hello @rflorent 

I manually tested all the modules against PrestaShop 8, and I didn't find any issues with `demodoctrine`. Could you please be more precise about what's wrong with this module on your side?

**Comment by rflorent** [comment]
Hello @kpodemski, @leemyongpakvn,

Thanks for your tests. 
Installation (+ @leemyongpakvn bug) : 
`10:06:50 ERROR     [app] Cannot register tab "AdminDemodoctrineQuote" because it already exists
10:06:50 ERROR     [app] Cannot register tab "AdminDemodoctrineQuote" because it already exists
10:06:50 ERROR     [app] Cannot register tab "AdminDemodoctrineQuote" because it already exists
10:06:50 ERROR     [app] Cannot register tab "AdminDemodoctrineQuote" because it already exists
10:06:50 ERROR     [app] Cannot register tab "AdminDemodoctrineQuote" because it already exists`

Package json is not up to date (webpack 2.7, no esbuild...), when trying to build assets : 
`ERROR in ./quotes/form.js
Module not found: Error: Can't resolve '@components/translatable-input' in '/xx/modules/demodoctrine/js/quotes'
 @ ./quotes/form.js 14:0-63

ERROR in ./quotes/index.js
Module not found: Error: Can't resolve '@components/grid/grid' in '/xx/modules/demodoctrine/js/quotes'
 @ ./quotes/index.js 10:0-41

ERROR in ./quotes/index.js
Module not found: Error: Can't resolve '@components/grid/extension/export-to-sql-manager-extension' in '/xx/modules/demodoctrine/js/quotes'
 @ ./quotes/index.js 12:0-101`

It's no more required to import core admin files as we can use `window.prestashop.component` : 
`import Grid from '@components/grid/grid';
import ReloadListActionExtension from '@components/grid/extension/reload-list-extension';
import ExportToSqlManagerExtension from '@components/grid/extension/export-to-sql-manager-extension';
import FiltersResetExtension from '@components/grid/extension/filters-reset-extension';
import SortingExtension from '@components/grid/extension/sorting-extension';
import LinkRowActionExtension from '@components/grid/extension/link-row-action-extension';
import SubmitGridExtension from '@components/grid/extension/submit-grid-action-extension';
import SubmitBulkExtension from '@components/grid/extension/submit-bulk-action-extension';
import BulkActionCheckboxExtension from '@components/grid/extension/bulk-action-checkbox-extension';
import SubmitRowActionExtension from '@components/grid/extension/action/row/submit-row-action-extension';`

Missing in admin template :
`{% form_theme pageForm '@PrestaShop/Admin/TwigTemplateForm/prestashop_ui_kit.html.twig' %}`
Result : I add a FormattedTextareaType and it isn't rendered correctly without the form_theme.
The field is unusable.

**Comment by kpodemski** [comment]
Hi @rflorent 

Thanks for reporting. I didn't compile assets, the rest worked fine for me. I'll get back to this module :)
